### ISSUE TITLE
GUI: ROI result is black

### ISSUE THREAD

**Issue by jhgee** [LiberTEM/LiberTEM#796]
When selecting a rectangle ROI which is too small the result stays black.

## works:
![grafik](https://user-images.githubusercontent.com/5411945/83244520-bc585900-a19f-11ea-8d22-40ddb49f83c8.png)
## doesn't work:
![grafik](https://user-images.githubusercontent.com/5411945/83244612-d6923700-a19f-11ea-8302-ffa60391728e.png)
## works:
![grafik](https://user-images.githubusercontent.com/5411945/83244739-02adb800-a1a0-11ea-8ec8-dd3307846957.png)
## doesn't work:
![grafik](https://user-images.githubusercontent.com/5411945/83244792-1c4eff80-a1a0-11ea-979e-3bf281f780fa.png)

**Comment by jhgee** [comment]
Same is true for disk ROIs

## works:
![grafik](https://user-images.githubusercontent.com/5411945/83245680-5a005800-a1a1-11ea-939b-5f00fddc993a.png)

## doesn't work:
![grafik](https://user-images.githubusercontent.com/5411945/83245727-6c7a9180-a1a1-11ea-8b5b-f24f0026e637.png)

**Comment by sk1p** [comment]
Is this using the master branch, or @AnandBaburajan's PR #793? Is this only happening with standard deviation, or also with average?

**Comment by jhgee** [comment]
@sk1p this happened indeed using the PR #793 revision https://github.com/LiberTEM/LiberTEM/pull/793/commits/391e8d498d417cfb92f12bb03e12acf8449533d3
It seems also depend on the dataset I loaded. With the current dataset I loaded I can't reproduce the issue but also got the zero devision error.
I will test average once I can reproduce the error again.

**Comment by jhgee** [comment]
@sk1p With the dataset from above also average leads to the same result. 
I thought about testing on the master branch version but this will not work, at leas with this unsynchronized dataset yet. I could try it with a synchronized one.

**Comment by sk1p** [comment]
> I could try it with a synchronized one.

That would be good - my guess is that region of interest handling also needs changes in #793

**Comment by jhgee** [comment]
I tested another dataset where even stuff like this works
![grafik](https://user-images.githubusercontent.com/5411945/83269984-2553c700-a1c8-11ea-831d-0b89dfae7596.png)

**Comment by jhgee** [comment]
seems not to be a problem in general
![grafik](https://user-images.githubusercontent.com/5411945/83270277-854a6d80-a1c8-11ea-8cae-4c3e20c088ce.png)

**Comment by AnandBaburajan** [comment]
> @AnandBaburajan could you have a look if this is related to your PR? Basically, making sure that regions of interest still work with your changes. That's also something that should get its own test case.

Yup, I'll look into this.

**Comment by AnandBaburajan** [comment]
> @AnandBaburajan could you have a look if this is related to your PR? Basically, making sure that regions of interest still work with your changes. That's also something that should get its own test case.

Yup, this was related to my PR. I've fixed it and I'll add tests after the implementation is complete.

> When selecting a rectangle ROI which is too small the result stays black.

This has been fixed.

> sometimes this is accompanied by:
> ![grafik](https://user-images.githubusercontent.com/5411945/83245086-8f587600-a1a0-11ea-8394-2c2773871502.png)

I never got this error though.

**Comment by sk1p** [comment]
> I never got this error though.

Did you test with standard deviation, as in the screenshot? If you decide to retry this and do manage to reproduce it, I would be interested in a traceback (should be on the console output) , as this could possibly be an unhandled boundary condition somewhere.

**Comment by AnandBaburajan** [comment]
> Did you test with standard deviation, as in the screenshot? If you decide to retry this and do manage to reproduce it, I would be interested in a traceback (should be on the console output) , as this could possibly be an unhandled boundary condition somewhere.

Sorry! I only tried the average mode. I just tried ROIs with standard deviation both on mine and the master branch and there's no result when the ROI is really small.

Regarding my PR, many rows of frames were giving no results in the average mode and that's been fixed.

But I still couldn't reproduce the `division by zero` error.

**Comment by sk1p** [comment]
> there's no result when the ROI is really small.

If the ROI only spans a single frame, this is expected, as the standard deviation of a single value should be zero, IIRC.

> Regarding my PR, many rows of frames were giving no results in the average mode and that's been fixed.

:+1:
